planner/core/rule_join_reorder_dp.go

```diff
@@ -122,11 +170,12 @@ func (s *joinReorderDPSolver) dpGraph(newPos2OldPos, oldPos2NewPos []int, joinGr
 				continue
 			}
 			// Get the edge connecting the two parts.
-			usedEdges := s.nodesAreConnected(sub, remain, oldPos2NewPos, totalEdges)
+			usedEdges, otherConds := s.nodesAreConnected(sub, remain, oldPos2NewPos, totalEqEdges, totalNonEqEdges)
+			// Here we only check equal condition currently.
```

Comment:
Why can't we treat `OtherConditions` as join connections now? Seems it can work already?